### ISSUE TITLE
Weird frame rate sometimes locked to 125fps even do just present with 3 backbuffers with vsync off

### ISSUE THREAD

**Issue by pengliu916** [Microsoft/DirectX-Graphics-Samples#61]
Hi, 
I have noticed some wired perf drop when running some of the samples with 3 backbuffer and present(0,0):  after keeping resizing the window for a while, the fps is locked to 125(8ms/frame) for a while. But when you have 5 backbuffer, this doesn't happen.

I have observed the same issue for D3D12HelloConstBuffers on multiple machines (change backbuffer count to 3, and use present(0,0) instead of present(1,0), and keeping resizing the window until that happen). And this even happen when we just have present(0,0) on OnRender() func.

One interesting thing I noticed is that whenever the fps lock happened, the dwm.exe cpu time drops from 7.5% to 5.2%. So I am not sure whether its the dwm causes that drop or it's the reverse way...

Anyway, this confuses me a lot, I have tried change the swapchain params, dxgi present modes, but non of them helps

**Comment by bobbrow** [comment]
I would recommend watching this video by Jesse Natalie of the D3D team:  https://www.youtube.com/watch?v=wn02zCXa9IU

TL;DR
Yes, you are being throttled by DWM.  If you switch to fullscreen mode (or add more back buffers), the throttling will go away.

In any case, if you pick up the Windows 10 update (aka: 1511, build 10586) that starting deploying about a week and a half ago, the throttling on windowed applications is removed for Present(0, 0) calls.

**Comment by pengliu916** [comment]
Thanks for the reply, but what interests me is why the frame is throttled about 8ms? Should it be 16.67ms for 60Hz which makes more sense? Also I am getting that result on 10586 build....

**Comment by bobbrow** [comment]
You are not throttled at 8 ms per frame.  You are throttled by the number of frames you try to Present.  So if you present 2 frames within 2 ms, your Present thread will be blocked for the remaining 14.67 ms.

Do you know how to use GPUView?  You can use that to visualize the frames you are attempting to present.  I can get about 2000 FPS on the HelloConstBuffers sample with an AMD R9 290 on the 10586 build.

**Comment by pengliu916** [comment]
Thanks clarification. I got some GPUView captures before, but can't fully understand the graph, also why the throttled framerate only happens sometimes?
